specutils/io/default_loaders/jwst_reader.py

```diff
@@ -585,6 +587,24 @@ def _jwst_s3d_loader(filename, **kwargs):
 
             wavelength = Quantity(wavelength_array, unit=wavelength_unit)
 
+            # The GWCS is currently broken for some IFUs, here we work around that
```

Comment:
Are you sure this affects GWCS and not the FITS WCS?

Reply:
Yes, the root issue is that the GWCS is specifying an array size that doesn't match the actual data array size along the spectral axis.